src/pages/Environments.js

```diff
@@ -313,6 +312,11 @@ const Environments = () => {
     })
   }
 
+  const multipleDisksError = (disks, appType) => {
+    return appType ? workspaceHasMultipleDisks(disks, appType) : _.remove(disk => getDiskAppType(disk) !== appType || disk.status === 'Deleting',
```

Comment:
We use `!!` to be explicit that we're coercing a non-boolean (`appType`) onto a boolean.

```suggestion
    return !!appType ? workspaceHasMultipleDisks(disks, appType) : _.remove(disk => getDiskAppType(disk) !== appType || disk.status === 'Deleting',
```

Reply:
Would `getDiskAppType(disk) !== appType` ever evaluate to `true` since it's in the second branch of the ternary where `!!appType` would be `false`?

Reply:
I think the code is iterating over all the disks, and we want to remove the ones that aren't associated with this specific appType.

Reply:
Right. I was trying to understand, for the case where `!!appType` is `false` (meaning, `appType` is `undefined`?), how `getDiskAppType(disk) !== appType` could be `true`. Looking more closely, it looks like `getDiskAppType(disk)` returns `undefined` if `disk` is unattached. In other words, `getDiskAppType(disk) !== appType` is necessary to filter out unattached disks?

Reply:
Not just unattached disks, but disks related to a different app. Like Galaxy vs. Cromwell.

Reply:
`_.remove(disk => getDiskAppType(disk) !== appType || disk.status === 'Deleting'…` part of the ternary would not be executed if a disk has `appType === Galaxy/Cromwell`.

Reply:
plan on discussing this comment on a call at noon today with @kyuksel if you'd like to join @cahrens

Reply:
For posterity: Talked to Gabriela offline and sorted out my misunderstanding.